src/Components/Pages/Calculator.js

```diff
@@ -1,13 +1,208 @@
-import PropTypes from 'prop-types';
+import PropTypes from 'prop-types'
 import React from 'react'
+import WASMWrapper from '../../Backend/WASMWrapper'
+
+import CalculatorSettings from '../CalculatorHelpers/CalculatorSettings'
+import CalculatorInput from '../CalculatorHelpers/CalculatorInput'
+import CalculatorOutput from '../CalculatorHelpers/CalculatorOutput'
+
+import * as convert from 'convert-units'
+
+const Units = {
+    Inputs: {
+        MeVPerNucleum: 'MeV/nucl',
+        //        Mev: 'Mev',
+    },
+    Outputs: {
+        SmallScale: 'keV/μm',
+        LargeScale: 'MeV/cm',
+        MassStoppingPower: 'Mev*g/cm^2'
+
+    }
+}
+
+const OperationMode = {
+    Dynamic: 0,
+    Performance: 1,
+}
+
 
 class CalculatorComponent extends React.Component {
     static propTypes = {
         ready: PropTypes.bool.isRequired
     }
 
+    async componentDidMount() {
+        try {
+            const [programs, ions, materials] = await Promise.all([
+                this.wrapper.getPrograms(),
+                this.wrapper.getIons(this.state.program),
+                this.wrapper.getMaterials(this.state.program)
+            ])
+            const program = programs[0]
+            const material = materials[0]
+            const ion = ions[0]
```

Comment:
Is it a place where default values are being set ?

Reply:
These are the default values after recieveing the data from libdedx. True defaults are empty and are set in constructor of the class

Reply:
Can we switch to following defaults ?

![image](https://user-images.githubusercontent.com/7374733/146937802-4104d60a-98bb-4b11-a2d9-1490eb5897af.png)

This would probably require using different index than 0

Reply:
I will be trying to extract the whole logic of program/ion/material selection in the next PR. Can we do it then? It will be easier to do for plot and data in one place

Reply:
Sure !